packages/design-system-react/src/components/button-primary/ButtonPrimary.test.tsx

```diff
@@ -131,16 +131,18 @@ describe('ButtonPrimary', () => {
     render(
       <ButtonPrimary
         startIconName={IconName.AddSquare}
+        startIconProps={{ 'data-testid': 'start-icon' }}
         endIconName={IconName.AddSquare}
+        endIconProps={{ 'data-testid': 'end-icon' }}
       >
         With Icons
       </ButtonPrimary>,
     );
 
-    const icons = screen.getAllByRole('img');
-    expect(icons).toHaveLength(2);
-    expect(icons[0]).toHaveClass('mr-2'); // start icon
-    expect(icons[1]).toHaveClass('ml-2'); // end icon
+    const startIcon = screen.getByTestId('start-icon');
+    const endIcon = screen.getByTestId('end-icon');
+    expect(startIcon).toHaveClass('mr-2');
+    expect(endIcon).toHaveClass('ml-2');
   });
```

Comment:
Updating tests that used the role of image for icons in button tests in favor of test ids